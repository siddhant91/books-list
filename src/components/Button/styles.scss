@import '../../styles/main.scss';

.bokl-button-component {
	border-radius: $bokl-spacing-1;
	border: 1px solid $bokl-color-purple;
	//  text-align: center;
	padding: $bokl-spacing-2;
	font-size: $bokl-body-font-size-medium;
	font-family: $bokl-font-family-secondary;
	font-weight: $bokl-font-weight-bold;
	cursor: pointer;
	// display: flex;
	// justify-content: center;
	// align-items: center;
	min-width: 200px;

	&.primary {
		background-color: $bokl-color-purple;
		color: $bokl-color-white;
		.bokl-icon {
			path {
				fill: $bokl-color-white;
			}
		}
	}
	&.secondary {
		color: $bokl-color-purple;
		background-color: $bokl-color-white;
	}
	&.disabled {
		border-color: $bokl-color-grey-5;
		background-color: $bokl-color-grey-5;
		color: $bokl-color-grey-3;
		cursor: not-allowed;
	}

	&.large {
		padding: $bokl-spacing-3 20px;
	}
	&.medium {
		padding: $bokl-spacing-2;
	}
	&.small {
		padding: 6px 20px;
		min-width: 100px;
	}
}
