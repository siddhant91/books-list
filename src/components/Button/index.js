/**
 * Generic Button Component
 */
import { node, string, bool, func, oneOf } from 'prop-types';
import clsx from 'clsx';

// Styles
import './styles.scss';

function Button({
	additionalClasses,
	appearance,
	size,
	children,
	disabled,
	handleClick,
	...props
}) {
	return (
		<button
			type="button"
			className={clsx(additionalClasses, appearance, size, 'bokl-button-component', {
				disabled,
			})}
			disabled={disabled}
			onClick={handleClick}
			data-testid="button"
			{...props}
		>
			{children}
		</button>
	);
}

Button.propTypes = {
	children: node.isRequired,
	disabled: bool,
	handleClick: func.isRequired,
	additionalClasses: string,
	appearance: oneOf(['primary', 'secondary', 'link', 'just-icon']),
	size: oneOf(['medium', 'large', 'small', 'icon']),
};

Button.defaultProps = {
	disabled: false,
	additionalClasses: '',
	appearance: 'primary',
	size: 'medium',
};

export default Button;
