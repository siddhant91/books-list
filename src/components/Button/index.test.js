import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';

import Button from './index';

afterEach(cleanup);

const fn = jest.fn();
const renderComponent = ({ disabled = false } = {}) => {
	return (
		<Button handleClick={fn} disabled={disabled}>
			Click Me
		</Button>
	);
};
it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent());
	expect(asFragment()).toMatchSnapshot();
});

it('Renders children props', () => {
	const { getByText } = render(renderComponent());
	expect(getByText('Click Me')).toBeInTheDocument();
});

it('Handles Button Click', async () => {
	const { getByTestId } = render(renderComponent());
	const button = getByTestId('button');
	fireEvent.click(button);
	await waitFor(() => {
		expect(fn).toHaveBeenCalledTimes(1);
	});
});

it('disables the Button', async () => {
	const { getByTestId } = render(renderComponent({ disabled: true }));
	const button = getByTestId('button');
	fireEvent.click(button);
	await waitFor(() => {
		expect(fn).toHaveBeenCalledTimes(0);
	});
});
