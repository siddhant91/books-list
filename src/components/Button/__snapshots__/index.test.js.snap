// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="primary medium bokl-button-component"
    data-testid="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
